adam/visualization/positioning.py

```diff
@@ -0,0 +1,426 @@
+                [-1, 1, 1],
+                [1, 1, 1],
+            ],
+            dtype=torch.float,
+        ).matmul(self.scale)
+        # see https://github.com/pytorch/pytorch/issues/24807 re: pylint issue
+
+    # helper functions giving names to a few corners used in calculations:
+    def right_corner(self) -> torch.Tensor:
+        """
+        Corner in the direction of the positive x axis from center (negative direction from other axes).
+        (Assuming box is oriented to world axes)
+        Returns: Tensor (3,)
+        """
+        return self.center + torch.tensor(  # pylint: disable=not-callable
+            [1, -1, -1], dtype=torch.float
+        ).matmul(self.scale)
+        # see https://github.com/pytorch/pytorch/issues/24807 re: pylint issue
+
+    def forward_corner(self) -> torch.Tensor:
+        """
+        Corner in the direction of the positive y axis from center (negative direction from other axes).
+        (Assuming box is oriented to world axes)
+        Returns: Tensor (3,)
+
+        """
+        return self.center + torch.tensor(  # pylint: disable=not-callable
+            [-1, 1, -1], dtype=torch.float
+        ).matmul(self.scale)
+
+    def up_corner(self) -> torch.Tensor:
+        """
+        Corner in the direction of the positive z axis from center (negative direction from other axes).
+        (Assuming box is oriented to world axes)
+        Returns: Tensor (3,)
+
+        """
+        return self.center + torch.tensor(  # pylint: disable=not-callable
+            [-1, -1, 1], dtype=torch.float
+        ).matmul(self.scale)
+
+    def minus_ones_corner(self) -> torch.Tensor:
+        """
+        Corner in the direction of the negative x, y, z axes from center
+        Returns: Tensor (3,)
+
+        """
+        return self.center + torch.tensor(  # pylint: disable=not-callable
+            [-1, -1, -1], dtype=torch.float
+        ).matmul(self.scale)
+
+    # functions returning normal vectors from three perpendicular faces of the box
+    def right_face(self) -> torch.Tensor:
+        """
+        Normal vector for the right face of the box (toward positive x axis when aligned to world axes)
+        Returns: Tensor (3,)
+        """
+        diff = self.right_corner() - self.minus_ones_corner()
+        return diff / torch.norm(diff)
+
+    def forward_face(self) -> torch.Tensor:
+        """
+        Normal vector for the forward face of the box (toward positive y axis when aligned to world axes)
+        Returns: Tensor (3,)
+        """
+        diff = self.forward_corner() - self.minus_ones_corner()
+        return diff / torch.norm(diff)
+
+    def up_face(self) -> torch.Tensor:
+        """
+        Normal vector for the up face of the box (toward positive z axis when aligned to world axes)
+        Returns: Tensor (3,)
+        """
+        diff = self.up_corner() - self.minus_ones_corner()
+        return diff / torch.norm(diff)
+
+    def get_face_norms(self) -> torch.Tensor:
```

Comment:
`face_normal_vectors` could be better, since "norm" has its own, different meaning.  I think you can drop the `get_`.